ostd/src/arch/loongarch/mod.rs

```diff
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: MPL-2.0
 
 //! Platform-specific code for the LoongArch platform.
+#![cfg_attr(target_arch = "loongarch64", expect(dead_code))]
```

Comment:
There is no need to use `cfg_attr(target_arch)` in `src/arch/*/mod.rs`. Use `expect(dead_code)` instead.